exporter/exportertest/mock_receiver.go

```diff
@@ -0,0 +1,237 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package exportertest // import "go.opentelemetry.io/collector/exporter/exportertest"
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+var errNonPermanent = status.Error(codes.DeadlineExceeded, "non Permanent error")
+var errPermanent = status.Error(codes.Internal, "Permanent error")
+
+type DecisionFunc func() error
+
+type MockReceiver interface {
+	Start() error
+	Stop() error
+	RequestCounter() RequestCounter
+}
+
+type MockReceiverFactory func(DecisionFunc) MockReceiver
```

Comment:
I think we can avoid exposing `DecisionFunc` by changing this signature to 
```suggestion
type MockReceiverFactory func(MockConsumer) MockReceiver
```

CreateDefaultConsumer will not be needed after that

Reply:
I changed it this way, but I think we still need a way to create MockConsumer to pass decision function based on test suite, so I created `NewMockConsumer`:

```
func NewMockConsumer(decisionFunc func() error) MockConsumer {
	return &mockConsumer{
		reqCounter:          newRequestCounter(),
		mux:                 sync.Mutex{},
		exportErrorFunction: decisionFunc,
		ReceivedTraces:      nil,
		ReceivedMetrics:     nil,
		ReceivedLogs:        nil,
	}
}
```

Didn't figure out how to create `MockConsumer` without an exported function. Let me know if I misunderstood 🙏